velox/exec/fuzzer/PrestoQueryRunner.cpp

```diff
@@ -749,6 +798,60 @@ std::string PrestoQueryRunner::createTable(
   return tableDirectoryPath;
 }
 
+std::unordered_map<std::string, std::vector<velox::RowVectorPtr>>
+PrestoQueryRunner::getAllTablesAndNames(const core::PlanNodePtr& plan) {
+  std::unordered_map<std::string, std::vector<velox::RowVectorPtr>> result;
+  if (const auto valuesNode =
+          std::dynamic_pointer_cast<const core::ValuesNode>(plan)) {
+    result.insert({getTableName(valuesNode), valuesNode->values()});
+  } else {
+    for (const auto& source : plan->sources()) {
+      auto tablesAndNames = getAllTablesAndNames(source);
+      result.insert(tablesAndNames.begin(), tablesAndNames.end());
+    }
+  }
+  return result;
+}
+
+std::optional<std::vector<velox::RowVectorPtr>>
+PrestoQueryRunner::executeVector(const core::PlanNodePtr& plan) {
+  if (const auto sql = toSql(plan)) {
```

Comment:
Why do we move `if (const auto sql = toSql(plan))` inside this method? It used to be called by caller (e.g., JoinFuzzer) before executeVector() or execute() is called. Was there any issue with that?

Reply:
IMO, this refactor makes it less error prone. It prevents invalid sql strings from being used. It also simplifies the function call to execute.

Reply:
This would cause redundant checks in aggregation and window fuzzers though, because we check toSql() separately from calling execute() to get fine-grained statistics (e.g., number of iterations not supported in reference DB vs. number of iterations fails in reference DB). See computeReferenceResults() in FuzzerUtil.cpp

Reply:
Oh I see, I didnt know that. Ok I reverted that change.